### DISCUSSION TITLE
Hope you can recommend me more datasets

### DISCUSSION THREAD

**Discussion by code-orangemonster** [facebookresearch/faiss#2838] [Q&A]
Hi, I am a master's student studying nearest neighbor search algorithm. At present, we need more datasets to test the effectiveness of the algorithm. I hope you can recommend me more dataset links and other information. You can contact me at “tragrace0125@gmail.com”. Thank you very much!

**Comment by GioiaZheng** [comment]
A few standard sources for nearest-neighbor benchmarking, by scale:

**Million-scale, ready-made with ground truth:**
- **ANN-Benchmarks** (`erikbern/ann-benchmarks`) — HDF5 files with precomputed exact neighbors: SIFT-128, GIST-960, GloVe (25/50/100/200d), Fashion-MNIST-784, NYTimes-256, Last.fm-65, DEEP-image-96. Drop-in for recall/QPS curves.
- **TEXMEX / corpus-texmex.irisa.fr** — the canonical SIFT1M, GIST1M, plus ANN_SIFT10K for quick tests.

**Billion-scale:**
- **big-ann-benchmarks** (NeurIPS'21 & '23 tracks) — BIGANN (SIFT1B), Microsoft SPACEV-1B, Microsoft Turing-ANNS-1B, Yandex DEEP-1B, Yandex Text-to-Image-1B, Facebook SimSearchNet++. All come with query sets and ground truth and are what current FAISS / large-scale ANN papers report on.

**Text-embedding (if you want semantic rather than synthetic vectors):**
- Encode **MS MARCO passage** or the **BEIR** corpora with a sentence encoder to get realistic high-dimensional distributions instead of SIFT/GIST.

For algorithm-effectiveness work, the ANN-Benchmarks suite is usually the fastest starting point because the ground truth and the evaluation harness are already provided.

**Comment by thlurte** [comment]
> Hi, I am a master's student studying nearest neighbor search algorithm. At present, we need more datasets to test the effectiveness of the algorithm. I hope you can recommend me more dataset links and other information. You can contact me at “[tragrace0125@gmail.com](mailto:tragrace0125@gmail.com)”. Thank you very much!

To add to the above section, if you are testing an algorithm for modern production systems, it is highly recommended to benchmark against the **four major workloads** that dominate current Approximate Nearest Neighbor (ANN) research:

### 1. Filtered Search (FANNS)
Real-world vector search almost always includes metadata filters (e.g., "find the nearest neighbor *where category = clothing and price < $50*"). Many fast algorithms fail when hard filters are applied because the vector space becomes fragmented.
* **YFCC-10M (NeurIPS 2023 Big-ANN):** 10 million CLIP embeddings of Flickr images paired with tag constraints (camera type, country, year).
* **Qdrant ANN Filtering Benchmark:** A dedicated GitHub repository (`qdrant/ann-filtering-benchmark-datasets`) containing multiple datasets specifically designed for filtered search.

### 2. Streaming and Dynamic Vectors
Modern vector databases must handle continuous insertions, deletions, and updates without degrading search recall or rebuilding the entire index.
* **MS Turing 10M / 30M (Streaming Track):** Released for the NeurIPS 2023 challenge, this includes a "runbook"—a sequence of insertion, deletion, and search operations. It tests how well an algorithm manages memory and maintains its graph/tree structure over time.

### 3. Out-of-Distribution (OOD) & Cross-Modal
In real systems, the distribution of query vectors often doesn't match the distribution of the base database (e.g., querying an image database using text embeddings). Algorithms that overfit to the base distribution often fail here.
* **Yandex Text-to-Image 10M:** A cross-modal dataset where the base 10M vectors are image embeddings, but the query vectors are textual search queries embedded by a different language model.

### 4. Sparse and Hybrid Search
Modern retrieval-augmented generation (RAG) systems combine dense distance metrics with high-dimensional sparse vectors (like BM25 or SPLADE) calculated via Maximum Inner Product Search (MIPS). 
* **MSMARCO/SPLADE:** An 8.8 million vector dataset where vectors have up to 100,000 dimensions, but only ~120 non-zero elements per vector. Graph-based ANN algorithms (like HNSW) behave very differently on sparse data compared to dense data.
